packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -182,6 +183,16 @@ export const parseRules = (rules: string): RulesParameters | null => {
   return { space, quorum, votingPeriod };
 };
 
+// Try parsing IPFS hash to validate it. This should also protect against replay attacks with different CIDv1 casing.
```

Comment:
How would this protect against replay attacks? Would a v1 hash fail?

Reply:
CID parser accepts only lower case for v1, so any attempts to manipulate by upper casing the same CID would throw and catch should return false

Reply:
Note that RFC 4648 does not strictly forbid mixed casing in base32 (used in CIDv1), but the implementation of multiformats library used in parsing does not allow mixed casing and defaults to lower-case alphabet when codec function is not passed to parse method.

Reply:
I totally get what you're saying about preventing the casing from allowing two v1 proposals to reference the same document but not be seen as a replay.

Just to make sure I understand... it's _not_ possible to reference the same IPFS document using CIDv1 and v2? Otherwise, couldn't you use that to replay?

Second question: do we have any language in the UMIP or ancillary data that specifies that only lower-casing for CIDv1 is allowed? Otherwise, if we disputed based on bad casing because of this failed validation, the dispute would be invalid, no?

Reply:
> Just to make sure I understand... it's not possible to reference the same IPFS document using CIDv1 and v2? Otherwise, couldn't you use that to replay?

Yes, it is possible to represent the same hash in CIDv0 and CIDv1 notation, but Snapshot backend stores single ipfs string property in the proposal, so any attempts to replay would be detected as only one version representations would match.

> Second question: do we have any language in the UMIP or ancillary data that specifies that only lower-casing for CIDv1 is allowed? Otherwise, if we disputed based on bad casing because of this failed validation, the dispute would be invalid, no?

No, this uses generic ASSERT_TRUTH identifier. [UMIP-170](https://github.com/UMAprotocol/UMIPs/blob/master/UMIPs/umip-170.md) instructs voters to check claim from emitted AssertionMade event. oSnap module contract adds rules to the claim that in the standard setup should include following language: "The explanation included with the on-chain proposal must be the unique IPFS identifier for the specific Snapshot proposal that was approved". So we might want to add explicit requirement to use lower case alphabet in CIDv1